src/docfx/lib/git/GitRepoInfoProvider.cs

```diff
@@ -2,16 +2,60 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
+using System.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class GitRepoInfoProvider
     {
-        private readonly ConcurrentDictionary<string, GitRepoInfo> _cache
+        private readonly ConcurrentDictionary<string, GitRepoInfo> _folderRepoInfocache
             = new ConcurrentDictionary<string, GitRepoInfo>();
 
+        private readonly ConcurrentDictionary<string, List<GitCommit>> _fileCommitsCache;
+
+        private readonly GitUserProfileCache _githubUserProfileCache = null;
```

Comment:
this should always be non-null. per https://github.com/dotnet/docfx/issues/2812, even if a cache file is not explicitly specified, we should use one at *%USERPROFILE%/.docfx/cache/user-profile.json* from run to run.